pageserver/src/tenant.rs

```diff
@@ -2208,12 +2208,20 @@ impl Tenant {
             let mut tuple = inspect_state(&rx.borrow_and_update());
 
             let is_broken = tuple.1;
-            if !is_broken {
+            let mut counted_broken = if !is_broken {
                 // the tenant might be ignored and reloaded, so first remove any previous set
                 // element. it most likely has already been scraped, as these are manual operations
                 // right now. most likely we will add it back very soon.
                 drop(crate::metrics::BROKEN_TENANTS_SET.remove_label_values(&[&tid]));
-            }
+                false
+            } else {
+                // add the id to the set right away, there should not be any updates on the channel
+                // after
```

Comment:
```suggestion
                // add the id to the set right away, there should not be any updates on the channel
                // after, because `Broken` is a terminal state
```

Reply:
We have the terminalness documented elsewhere don't we?

Reply:
https://github.com/neondatabase/neon/blob/77a68326c5bf957e905654c69642da3caca649aa/libs/pageserver_api/src/models.rs#L86-L95 has catalogue of where it's called, I think that is good enough.

I don't think it's a good idea to keep leaving which aim at cataloging what exists vs. why; they will go stale.

Reply:
Disagree, when we inevitable implement "recover from Broken" state in the future, someone is going to have to inspect this code here and they're going to wonder whether ` because `Broken` is a terminal state` was exactly the reason why there wouldn't be `any updates on the channel after`, or whether there was something else to it.

But whatever